### PR TITLE
Prevent transferring working copies in remote task.

### PR DESCRIPTION
**Pull request by deiferni**
Prevent transferring documents checked out by different users when completing successor tasks in `@complete-successor-task`. This makes the behavior consistent with validation in the classic UI where such documents are not allowed.

Screenshot 2021-09-08 at 14 56 33

Error handling is unfortunately very basic for the new UI but we have other issues that block having something better right now.

For [CA-2156](https://4teamwork.atlassian.net/browse/CA-2156)

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- API change:
  - [x] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))

### CODE REVIEW

File: changes/CA-2156-2.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Prevent transferring documents checked out by different users when completing successor tasks. [deiferni]
```
Comment (njohner) [inline changes/CA-2156-2.bugfix:1]: From what I see it does not matter if I checked it out myself or some other user. So I'd rephrase that to:
"Prevent transferring checked out documents when completing successor tasks. [deiferni]"

File: docs/public/dev-manual/api/api_changelog.rst [hunk 2]
```diff
@@ -22,6 +22,7 @@ Other Changes
 - Prevent setting ``file`` to ``null`` for ``opengever.document.document`` if it is inside an ``opengever.meeting.proposal``.
 - Include checkout collaborators and file modification time in document serialization.
 - Include checkout collaborators, file modification time, lock time and lock timeout in document status.
+- ``@complete-successor-task``: Prevent transferring documents checked out by different users.
```
Comment (njohner) [inline docs/public/dev-manual/api/api_changelog.rst:25]: same here

File: opengever/api/complete_successor_task.py [hunk 3]
```diff
@@ -71,6 +72,10 @@ def reply(self):
         # Map any supported document reference type to an IntId
         docs_to_deliver = map(self._resolve_doc_ref_to_intid, docs_to_deliver)
 
+        invalid = get_checked_out_documents(docs_to_deliver)
+        if invalid:
+            raise BadRequest('Some documents are checked out by a different user.')
```
Comment (njohner) [inline opengever/api/complete_successor_task.py:77]: and here

**Review by njohner — APPROVED** [review]
See comment, otherwise LGTM

File: changes/CA-2156-2.bugfix (hunk 1 above)
Comment (deiferni) [reply changes/CA-2156-2.bugfix:1]: Thanks! This is what i first implemented but then i saw that the classic ui always blocks, even if the current user has the checkout. amended!
